sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -457,7 +457,7 @@ object DataType {
 
       case (w: AtomicType, r: AtomicType) if storeAssignmentPolicy == STRICT =>
         if (!Cast.canUpCast(w, r)) {
-          addError(s"Cannot safely cast '$context': $w to $r")
+          addError(s"Cannot safely cast '$context': ${w.simpleString} to ${r.simpleString}")
```

Comment:
nit: `simpleString` -> `catalogString`?

Reply:
+1 for `catalogString `:
https://github.com/apache/spark/blob/0fd98abd859049dc3b200492487041eeeaa8f737/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L268-L269